![Parked motorcycle](motorcycle.jpg)

You're working for a company that sells motorcycle parts, and they've asked for some help in analyzing their sales data!

They operate three warehouses in the area, selling both retail and wholesale. They offer a variety of parts and accept credit cards, cash, and bank transfer as payment methods. However, each payment type incurs a different fee.

The board of directors wants to gain a better understanding of wholesale revenue by product line, and how this varies month-to-month and across warehouses. You have been tasked with calculating net revenue for each product line and grouping results by month and warehouse. The results should be filtered so that only `"Wholesale"` orders are included.

They have provided you with access to their database, which contains the following table called `sales`:

## Sales
| Column | Data type | Description |
|--------|-----------|-------------|
| `order_number` | `VARCHAR` | Unique order number. |
| `date` | `DATE` | Date of the order, from June to August 2021. |
| `warehouse` | `VARCHAR` | The warehouse that the order was made from&mdash; `North`, `Central`, or `West`. |
| `client_type` | `VARCHAR` | Whether the order was `Retail` or `Wholesale`. |
| `product_line` | `VARCHAR` | Type of product ordered. |
| `quantity` | `INT` | Number of products ordered. | 
| `unit_price` | `FLOAT` | Price per product (dollars). |
| `total` | `FLOAT` | Total price of the order (dollars). |
| `payment` | `VARCHAR` | Payment method&mdash;`Credit card`, `Transfer`, or `Cash`. |
| `payment_fee` | `FLOAT` | Percentage of `total` charged as a result of the `payment` method. |


Your query output should be presented in the following format:

| `product_line` | `month` | `warehouse` |	`net_revenue` |
|----------------|-----------|----------------------------|--------------|
| product_one | --- | --- | --- |
| product_one | --- | --- | --- |
| product_one | --- | --- | --- |
| product_one | --- | --- | --- |
| product_one | --- | --- | --- |
| product_one | --- | --- | --- |
| product_two | --- | --- | --- |
| ... | ... | ... | ... |

In [4]:
-- Start coding here
-- Start coding here
SELECT product_line,
    CASE WHEN EXTRACT('month' from date) = 6 THEN 'June'
        WHEN EXTRACT('month' from date) = 7 THEN 'July'
        WHEN EXTRACT('month' from date) = 8 THEN 'August'
    END as month,
    warehouse,
	SUM(total) - SUM(payment_fee) AS net_revenue
FROM sales
WHERE client_type = 'Wholesale'
GROUP BY product_line, warehouse, month
ORDER BY product_line, month, net_revenue DESC

,product_line,month,warehouse,net_revenue
0,Braking system,August,Central,3039.41
1,Braking system,August,West,2500.67
2,Braking system,August,North,1770.84
3,Braking system,July,Central,3778.65
4,Braking system,July,West,3060.93
5,Braking system,July,North,2594.44
6,Braking system,June,Central,3684.89
7,Braking system,June,North,1487.77
8,Braking system,June,West,1212.75
9,Electrical system,August,North,4721.12


Additional Independent Business Analysis

To further explore the company’s sales performance, I analyzed warehouse revenue, product performance, monthly trends, and payment fees.

### Which warehouse generated the most total net wholesale revenue?

In [2]:
SELECT
    warehouse,
    SUM(total) - SUM(payment_fee) AS net_revenue
FROM sales
WHERE client_type = 'Wholesale'
GROUP BY warehouse
ORDER BY net_revenue DESC;

,warehouse,net_revenue
0,Central,78855.67
1,North,58065.50
2,West,22718.91


### Findings
The Central warehouse generated the highest net wholesale revenue at $78,855.67 , followed by North at $58,065.50 and West at $22,718.91. Central generated more than three times the net wholesale revenue of West, showing a substantial difference in sales performance across warehouse locations.

,warehouse,net_revenue
0,Central,78855.67
1,North,58065.50
2,West,22718.91


### Which product line generated the most net wholesale revenue overall?

In [6]:
SELECT
    product_line,
    SUM(total) - SUM(payment_fee) AS net_revenue
FROM sales
WHERE client_type = 'Wholesale'
GROUP BY Product_line
ORDER BY net_revenue DESC;

,product_line,net_revenue
0,Frame & body,39477.51
1,Suspension & traction,38294.76
2,Braking system,23130.35
3,Electrical system,21754.03
4,Engine,21235.92
5,Miscellaneous,15747.51


### Findings
Frame & body generated the highest net wholesale revenue at $39,477.51, closely followed by Suspension & traction at $38,294.76. Together, these two product lines substantially outperformed the remaining categories, with revenue dropping to $23,130.35 for Braking system, the third-highest category. This suggests that Frame & body and Suspension & traction are particularly important contributors to the company’s wholesale revenue.

,product_line,net_revenue
0,Frame & body,39477.51
1,Suspension & traction,38294.76
2,Braking system,23130.35
3,Electrical system,21754.03
4,Engine,21235.92
5,Miscellaneous,15747.51


### How did net wholesale revenue change from June to July to August?

In [10]:
SELECT
    EXTRACT('month' FROM date) AS month,
    SUM(total) - SUM(payment_fee) AS net_revenue
FROM sales
WHERE client_type = 'Wholesale'
GROUP BY month
ORDER BY month ASC;

,month,net_revenue
0,6,49099.18
1,7,49085.35
2,8,61455.55


### Findings
Net wholesale revenue remained nearly unchanged from June ($49,099.18) to July ($49,085.35). In August, revenue increased substantially to $61,455.55, representing an increase of approximately 25% compared with July. This indicates a significant improvement in wholesale sales performance during August and suggests further analysis could identify which products or warehouses contributed most to the increase.

,month,net_revenue
0,6,49099.18
1,7,49085.35
2,8,61455.55


### What caused the August revenue increase—which product lines contributed most to the growth?

In [16]:
SELECT
    product_line,

    SUM(CASE
        WHEN EXTRACT('month' FROM date) = 7
        THEN total - payment_fee
        ELSE 0
    END) AS july_revenue,

    SUM(CASE
        WHEN EXTRACT('month' FROM date) = 8
        THEN total - payment_fee
        ELSE 0
    END) AS august_revenue

FROM sales

WHERE client_type = 'Wholesale'
    AND EXTRACT('month' FROM date) IN (7, 8)

GROUP BY product_line

ORDER BY product_line;

,product_line,july_revenue,august_revenue
0,Braking system,9434.02,7310.92
1,Electrical system,7737.21,9089.39
2,Engine,2834.17,11852.90
3,Frame & body,9289.74,17386.57
4,Miscellaneous,6679.89,4394.59
5,Suspension & traction,13110.32,11421.18


### Findings
The increase in August net wholesale revenue was driven primarily by strong growth in the Engine and Frame & body product lines. Engine revenue increased from about $2,834 in July to $11,853 in August, while Frame & body increased from about $9,290 to $17,387. Electrical system also increased modestly, while Braking system, Miscellaneous, and Suspension & traction declined. This suggests that the overall August revenue increase was concentrated in a few high-growth product categories rather than across all product lines.

,product_line,month,net_revenue
0,Braking system,7,9434.02
1,Braking system,8,7310.92
2,Electrical system,7,7737.21
3,Electrical system,8,9089.39
4,Engine,7,2834.17
5,Engine,8,11852.90
6,Frame & body,7,9289.74
7,Frame & body,8,17386.57
8,Miscellaneous,7,6679.89
9,Miscellaneous,8,4394.59


### How did payment fees affect total wholesale revenue?

In [17]:
SELECT
    payment,
    SUM(payment_fee) AS total_payment_fees,
    SUM(total) AS gross_revenue,
    SUM(total) - SUM(payment_fee) AS net_revenue
FROM sales
WHERE client_type = 'Wholesale'
GROUP BY payment
ORDER BY total_payment_fees DESC;

,payment,total_payment_fees,gross_revenue,net_revenue
0,Transfer,2.25,159642.33,159640.08


### Findings
Wholesale transactions generated $159,642.33 in gross revenue and incurred only $2.25 in payment fees, resulting in $159,640.08 in net revenue. Since wholesale orders used bank transfers, payment fees had a minimal impact on overall wholesale revenue.